zilliqa/src/node.rs

```diff
@@ -119,10 +119,18 @@ impl Node {
         match message {
             Message::External(external_message) => match external_message {
                 ExternalMessage::Proposal(m) => {
+                    let m_view = m.header.view;
+
                     if let Some((leader, vote)) = self.consensus.proposal(m)? {
                         self.reset_timeout.send(())?;
                         self.message_sender
                             .send_external_message(leader, ExternalMessage::Vote(vote))?;
+                    } else if m_view % 5 == 0 {
```

Comment:
Maybe it makes sense to keep this logic inside consensus instead? It feels a bit magic here to trigger this behaviour based on not responding to a proposal.

Reply:
This would also mean that you could send multiple messages and/or of different types when you recieve one - for example sending JoinCommittee instead of Vote when you see a block proposal but you are not part of committee